src/DistrictResources/DistrictResources.cpp

```diff
@@ -1,4 +1,277 @@
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <valarray>
 #include "DistrictResources.h"
+#include "../Vehicle/FireEngine.h"
+#include "../Vehicle/FireLadder.h"
+#include "../Vehicle/Vehicle.h"
+#include "../Event/Event.h"
 
 namespace LockFreeDispatch {
+
+    void DistrictResources::populateDistrictVehicles(const std::vector<Vehicle>& vehicles)
+    {
+        for (auto vehicle : vehicles)
+        {
+            districtVehicles.push_back(vehicle);
+        }
```

Comment:
```suggestion
        for (const auto & vehicle : vehicles)
        {
            districtVehicles.push_back(vehicle);
        }
```
`push_back()` creates a copy anyway, so it might be more efficient to use `const&` than the copy constructor in this loop. Your `Vehicle` class looks relatively expensive to copy.

Reply:
For loops below look analogous.